### ISSUE TITLE
Add troubleshooting note on Linux

### ISSUE THREAD

**Issue by CraftedCart** [aldelaro5/Dolphin-memory-engine#25]
On Linux, the kernel will deny access to read/write another process' memory by default. The easiest way to get around this (without enabling it for all processes) is to run `setcap cap_sys_ptrace=eip path/to/Dolphin-memory-engine`.

Just thought you might want to mention that on the README.md and/or perhaps within the application, else the program appears to immediately unhook after hooking.

**Comment by aldelaro5** [comment]
This is odd, can I have a link or something that explain this in details?

The reason I ask this is because on my arch linuc, I haven't edited the kernel that the repository provides and this syscall worked perfectly on the first try, when I was doing a prototype.

Perhaps this is true with a particular version of the kenerl?

**Comment by CraftedCart** [comment]
Huh - heck I use Arch Linux also, with the stock kernel.

```
$ uname -r
4.15.6-1-ARCH
```

Just checked again (This time using DME both from the releases and building it myself, in case something recent happened (unlikely, but eh)) - yup still needed to `setcap`.

As for a link, I just used their Wiki section on Dwarf Therapist :P https://wiki.archlinux.org/index.php/Dwarf_Fortress#Dwarf_Therapist

---

Some more searching later, I found this: http://man7.org/linux/man-pages/man5/proc.5.html (As it appears, from a quick scan through your source code, that you just use `/proc/pid/maps` to access Dolphin's memory)

---

`/proc/[pid]/maps`

A file containing the currently mapped memory regions and
their access permissions.  See mmap(2) for some further infor‐
mation about memory mappings.

**Permission to access this file is governed by a ptrace access
mode PTRACE_MODE_READ_FSCREDS check; see ptrace(2).**

---

That man page then links to the ptrace(2) page: http://man7.org/linux/man-pages/man2/ptrace.2.html

---

The algorithm employed for ptrace access mode checking determines
       whether the calling process is allowed to perform the corresponding
       action on the target process.  (In the case of opening /proc/[pid]
       files, the "calling process" is the one opening the file, and the
       process with the corresponding PID is the "target process".)  The
       algorithm is as follows:

[...]

3. Deny access if neither of the following is true:

· The real, effective, and saved-set user IDs of the target match
the caller's user ID, and the real, effective, and saved-set
group IDs of the target match the caller's group ID.

**· The caller has the CAP_SYS_PTRACE capability in the user
namespace of the target.**

---

In this case, the caller DME does not have the CAP_SYS_PTRACE capability
